src/api.py

```diff
@@ -1,26 +1,100 @@
-from flask import Flask, request
+import os
+
+from celery.result import AsyncResult
+from fastapi import FastAPI
+from fastapi.params import Query
+from pydantic import BaseModel
+from starlette.responses import HTMLResponse
+
 from src import service
+from src.generators import html_generator
+
+app = FastAPI()
+format_html = False
+
 
-app = Flask(__name__)
+class TaskOut(BaseModel):
+    id: str
+    status: str
+    result: str | None = None
+    status_callback_url: str | None = None
 
 
 # Sample url: https://tmt.org/?test-url=https://github.com/teemtee/tmt&test-name=/tests/core/smoke
 # or for plans: https://tmt.org/?plan-url=https://github.com/teemtee/tmt&plan-name=/plans/features/basic
-@app.route("/", methods=["GET"])
-def find_test():
-    test_url = request.args.get("test-url", default=None)
-    test_name = request.args.get("test-name", default=None)
-    test_ref = request.args.get("test-ref", default="main")
+@app.get("/")
+def find_test(
+        test_url: str = Query(None, alias="test-url"),
```

Comment:
To better support static code analysis tools, it is preferred to use `Annotated`, eg.:

```suggestion
        test_url: Annotated[str | None, Query(alias="test-url")] = None,
```

And if you'd like to add more documentation for the query parameters:

```suggestion
        test_url: Annotated[
            str | None,
            Query(
                alias="test-url",
                title="Test URL",
                description="URL of a Git repository containing test metadata",
            ),
        ] = None,
```

References:
  * https://fastapi.tiangolo.com/tutorial/query-params-str-validations/#alias-parameters
  * https://github.com/fastapi/fastapi/discussions/9279
  * https://fastapi.tiangolo.com/tutorial/query-params-str-validations/#declare-more-metadata

Reply:
Awesome, thanks for this, I was just trying to debug this :)

Reply:
Need to add `from typing import Annotated`.

I've not added suggestions for all the parameters that should be changed like this (as there's a ton of them), but all of them should be modified.